cpp/src/arrow/filesystem/filesystem_test.cc

```diff
@@ -265,6 +265,33 @@ TEST(PathUtil, ToSlashes) {
 #endif
 }
 
+TEST(PathUtil, Globber) {
+  Globber empty("");
+  ASSERT_FALSE(empty.Matches("/1.txt"));
+
+  Globber star("/*");
+  ASSERT_TRUE(star.Matches("/a.txt"));
+  ASSERT_TRUE(star.Matches("/b.csv"));
+  ASSERT_FALSE(star.Matches("/foo/c.parquet"));
```

Comment:
Can we also add a quick test that `/a?b` does not match `/a/b`?

Reply:
Sure